### PR TITLE
feat: update to eslint9, use new plugin format

### PR DESCRIPTION
**Pull request by stabback**
We should be able to drop eslint-config-6river with this
https://eslint.org/docs/latest/extend/plugins#configs-in-plugins

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+//------------------------------------------------------------------------------
+// Plugin Definition
+//------------------------------------------------------------------------------
+
+const plugin = {
+	rules: requireIndex(__dirname + '/rules'),
+	meta: {
+		name: pkg.name,
+		// version: pkg.version,
```
Comment (mgabeler-lee-6rs) [inline lib/plugin.js]: Any reason not to uncomment this?

Comment (stabback) [reply lib/plugin.js]: Gotta get CI to update the package version, unless that happens in run-release?

Comment (mgabeler-lee-6rs) [reply lib/plugin.js]: I think it does, double-checking...

Comment (mgabeler-lee-6rs) [reply lib/plugin.js]: Yes, run-release will patch up package.json:
https://github.com/6RiverSystems/circleci-orbs/blob/b58df14fba8c988c9e18727e1305bd21cfb8e916/src/scripts/run-release.sh#L42-L43

Comment (stabback) [reply lib/plugin.js]: dope

**Review by mgabeler-lee-6rs — APPROVED** [review]

**Pull request merged**